File: frontend/src/types/journey.ts
export interface JourneyAlert {
  id: string;
  stage: string;
  severity: 'INFO' | 'WARNING' | 'CRITICAL' | 'SUCCESS';
  title: string;
  message: string;
  timestamp: string;
  action_label?: string;
  action_type?: string;
}

export interface StageChecklistItem {
  id: string;
  task: string;
  completed: boolean;
  required: boolean;
  tip: string;
}

export interface JourneyStageDetail {
  stage_id: 'admission' | 'treatment' | 'billing' | 'discharge';
  title: string;
  subtitle: string;
  status: 'COMPLETED' | 'IN_PROGRESS' | 'UPCOMING';
  metrics: Record<string, string>;
  checklist: StageChecklistItem[];
  alerts: JourneyAlert[];
}

export interface JourneyStatusResponse {
  patient_name: string;
  hospital_name: string;
  policy_name: string;
  admission_number: string;
  current_stage_id: 'admission' | 'treatment' | 'billing' | 'discharge';
  pre_auth_approved_amount: number;
  current_interim_bill: number;
  out_of_pocket_estimated: number;
  stages: JourneyStageDetail[];
  active_alerts: JourneyAlert[];
}

export interface ClaimDossierResponse {
  dossier_id: string;
  generated_at: string;
  patient_name: string;
  hospital_name: string;
  policy_number: string;
  total_bill: number;
  cashless_sanctioned: number;
  copay_settled: number;
  caregiver_paid: number;
  documents_checklist: string[];
  tpa_submission_code: string;
  summary_text: string;
}

File: backend/app/schemas/journey.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel

class JourneyAlert(BaseModel):
    id: str
    stage: str
    severity: str # "INFO", "WARNING", "CRITICAL", "SUCCESS"
    title: str
    message: str
    timestamp: str
    action_label: Optional[str] = None
    action_type: Optional[str] = None

class StageChecklistItem(BaseModel):
    id: str
    task: str
    completed: bool
    required: bool = True
    tip: str

class JourneyStageDetail(BaseModel):
    stage_id: str # "admission", "treatment", "billing", "discharge"
    title: str
    subtitle: str
    status: str # "COMPLETED", "IN_PROGRESS", "UPCOMING"
    metrics: Dict[str, Any]
    checklist: List[StageChecklistItem]
    alerts: List[JourneyAlert]

class JourneyStatusResponse(BaseModel):
    patient_name: str
    hospital_name: str
    policy_name: str
    admission_number: str
    current_stage_id: str
    pre_auth_approved_amount: float
    current_interim_bill: float
    out_of_pocket_estimated: float
    stages: List[JourneyStageDetail]
    active_alerts: List[JourneyAlert]

class AdvanceJourneyRequest(BaseModel):
    target_stage_id: str
    action: Optional[str] = None
    extra_data: Optional[Dict[str, Any]] = None

class ClaimDossierResponse(BaseModel):
    dossier_id: str
    generated_at: str
    patient_name: str
    hospital_name: str
    policy_number: str
    total_bill: float
    cashless_sanctioned: float
    copay_settled: float
    caregiver_paid: float
    documents_checklist: List[str]
    tpa_submission_code: str
    summary_text: str
